app/src/main/java/com/example/screentimemanager/usageComparisonService/UsageNotificationWorker.kt

```diff
@@ -0,0 +1,86 @@
+package com.example.screentimemanager.workers
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.example.screentimemanager.usageComparisonService.UsageComparisonManager
+import com.example.screentimemanager.data.firebase.usage.UsageFirebaseDao
+import com.example.screentimemanager.data.firebase.user.UserFirebaseDao
+import com.example.screentimemanager.R
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.FirebaseDatabase
+import java.util.Calendar
+import java.util.Date
+
+class UsageNotificationWorker(
+    context: Context,
+    params: WorkerParameters,
+) : CoroutineWorker(context, params) {
+
+    override suspend fun doWork(): Result {
+        var databaseReference: DatabaseReference = FirebaseDatabase.getInstance().reference
+        val usageFirebaseDao = UsageFirebaseDao(databaseReference)
+        val userFirebaseDao = UserFirebaseDao(databaseReference)
+        val usageComparisonManager = UsageComparisonManager(usageFirebaseDao, userFirebaseDao)
+        // Try to get userEmail from inputData first
+        val inputDataEmail = inputData.getString("userEmail")
+
+        // Fallback to the currently logged-in user's email if inputDataEmail is not available
+        val userEmail = inputDataEmail ?:"abc123@gmail.com"
+
+        // If userEmail is still null, return failure
+        if (userEmail == null) {
+            Log.d("UsageNotificationWorker", "Notification In the worker class, userEmail $userEmail is null. Returning failure")
+            return Result.failure()
+        }
+
+        val c = Calendar.getInstance()
+        val year = c.get(Calendar.YEAR)
+        val month = c.get(Calendar.MONTH)
```

Comment:
calendar.get(Calendar.MONTH) is 0 based so when the month is 12 it'll return 11. Right it i believe its checking for the previous month. Theres a function in the Util class that you could also use instead.